components/dashboard/src/components/PrebuildLogs.tsx

```diff
@@ -112,13 +102,31 @@ export default function PrebuildLogs(props: PrebuildLogsProps) {
                         }
                         logsEmitter.emit("logs", content.text);
                     },
-                    onPrebuildUpdate(update: PrebuildWithStatus) {
-                        if (update.info) {
-                            handlePrebuildUpdate(update);
-                        }
-                    },
                 }),
             );
+
+            try {
+                const response = await prebuildClient.listPrebuilds({ buildWorkspaceId: props.workspaceId });
+                const prebuild = response.prebuilds[0];
+                if (prebuild) {
+                    handlePrebuildUpdate(prebuild);
```

Comment:
@svenefftinge @geropl Could it be that there is not yet prebuild for a build workspace id? It should not happen right?

The trouble with new code that it is less resilient, before even if we fail to fetch prebuild (for any reason, network, etc.) we still will listen for all updated and can recover. Now if we cannot fetch it UI will fail, only reload can be used to recover.

One solution could be to move build_workspace_id from List to Watch, and just call here watch. It will grab associated configuration, listen for all prebuild updates and if workspace id is matching propagate them.

Reply:
> Could it be that there is not yet prebuild for a build workspace id?

I don't think so, and if yes, it's a bug - and it can only happen until we moved to the new PrebuildApi, because that's the only way you know about that field.
I think that's not important enough to change the API.

Reply:
Sounds good to me. FWIW, I believe the field should be named `built_workspace_id` or rather just `workspace_id`. I don't find that ambiguous  on a prebuild to point to its workspace like that.

Reply:
> I believe the field should be named built_workspace_id or rather just workspace_id

build vs built is also a matter of perspective, and IMO already showed to be problematic (as in: I can never remember which field is named how and why :see_no_evil: ).
But yeah, the shorter the better.